JoeDodgson/Fitness-buddy
# Jacob/bugfix/heroku deployment failure

Large change so comb through this one closely:

* Changed some node modules from dev to normal and vice versa
* Added some comments through
  - please note in review if there's anything that's not clear so I can comment further

# Testing
* Read code line by line.
* Connect to JAWSDB from local workbench
* Test deployment
* Ensure that db is updating when being used on heroku

**If anything fails please let me know asap so I can check logs**

## express-session
* Used to create an active session for passport to use.
  - This uses cookies so may be worth looking into converting for session storage but that's for another day.
* Works by storing a session key in the DB that relates to the user that is logged in.
  - Have a play with logging in on the deployment and checking the changes in the DB that this causes
  - This is only usable in the dev environment, need to use an alternative storage for production. so using:

## express-mysql-session
* This simply handles access to the db.
  - Has it's own setting for session expiry along with passport, they work in tandem in a weird way.
  - Doesn't accept a URL for db so tried creating a function that parsed it, which both worked and didn't.
  - Opted to store values separately in env variables on heroku and access them that way.
